### PR TITLE
Fixed WF agent to work more reliably.

### PR DESCRIPTION
**Pull request by laurajogi**
Agent checks if wall is followed before when it has found a new wall while doing heuristics.
Basically uses a pheromone/marker idea to check if walls are followed before, but the pheromone smell (or anything like that) has not yet been implemented.

### CODE REVIEW

File: src/main/java/app/model/agents/WallFollow/WallFollowAgent.java [hunk 1]
```diff
@@ -546,6 +588,32 @@ public boolean noObstacleDetectedFromPos(double rayAngle, int distFromAgentVerte
...
+        }
+    }
+
+    private boolean foundUnexploredWallToFollow()
+    {
+        BooleanCell forwardCell = getAgentNeighbourBasedOnAngle(direction.getAngle());
+        if (!noWallDetected(direction.getAngle()) && forwardCell.getObstacle()
+                && ((!horizontalWallsCovered.contains(forwardCell.getX()) && direction.getX() == 0)
+                || (!verticalWallsCovered.contains(forwardCell.getY()) && direction.getY() == 0)))
+        {
```
Comment (S010MON) [inline src/main/java/app/model/agents/WallFollow/WallFollowAgent.java:611]: consider simplifying to :
```java

return (!noWallDetected(direction.getAngle()) && forwardCell.getObstacle()
           && ((!horizontalWallsCovered.contains(forwardCell.getX()) && direction.getX() == 0)
           || (!verticalWallsCovered.contains(forwardCell.getY()) && direction.getY() == 0)))
```
As returning a boolean evaluation will always give you the same as an if true/ else false

**Review by S010MON — APPROVED** [review]
Looks really good, one slight bug, which is that the agents don't seem to go through the portals.  Definitely ready to push, perhaps deal with that issue later.

**Comment by laurajogi** [comment]
Tested the WF agent on a simple map with many portals and they do go through the portals when they go directly towards it, i.e. the portal is in front of the agent. But when it's just to the left or right of them, they don't recognise it as a corner to turn around at and they go past it.

**Review by MatthewFoster02 — APPROVED** [review]
Great work.

**Review by S010MON — APPROVED** [review]

**Pull request merged**